pkg/controller/validate_config.go

```diff
@@ -104,3 +111,78 @@ func (c *ValidateConfigHandler) Handle(ctx context.Context) {
 	ctx = CtxClusterStatus.WithValue(ctx, currentStatus)
 	c.next.Handle(ctx)
 }
+
+func (c *ValidateConfigHandler) getAvailableVersions(ctx context.Context, graph updates.UpdateGraph, version v1alpha1.SpiceDBVersion, datastore string) []v1alpha1.SpiceDBVersion {
+	logger := logr.FromContextOrDiscard(ctx)
+	source, err := graph.SourceForChannel(version.Channel)
+	if err != nil {
+		logger.V(4).Error(err, "no source found for channel %q, can't compute available versions", version.Channel)
+	}
+
+	availableVersions := make([]v1alpha1.SpiceDBVersion, 0)
+	nextDirect := source.NextDirect(version.Name)
+	next := source.Next(version.Name)
```

Comment:
Maybe only compute `next` after the conditional below?

Reply:
we always need `next` even if the others return values

Reply:
its not used until after the next conditional?